sql/core/src/test/resources/sql-tests/inputs/array.sql

```diff
@@ -130,6 +130,18 @@ select get(array(1, 2, 3), 3);
 select get(array(1, 2, 3), null);
 select get(array(1, 2, 3), -1);
 
+-- function array_insert()
+select array_insert(array(1, 2, 3), 3, 4);
+select array_insert(array(2, 3, 4), 0, 1);
+select array_insert(array(2, 3, 4), 1, 1);
+select array_insert(array(1, 3, 4), -2, 2);
+select array_insert(array(1, 2, 3), 3, "4");
+select array_insert(cast(NULL as ARRAY<INT>), 1, 1);
+select array_insert(array(1, 2, 3, NULL), cast(NULL as INT), 4);
```

Comment:
There doesn't seem to be a way to specify that the array can have NULL elements in the raw SQL syntax. I've added a null to the base array in the meantime, but please let me know if there is a better way to avoid the following outcome:

-- !query
select array_insert(array(2, 3, 4), 5, 1)
-- !query schema
struct<array_insert(array(2, 3, 4), 5, 1):array>
-- !query output
[2,3,4,0,1]